### ISSUE TITLE
Don't process fields marked as "static" and/or "transient"

### ISSUE THREAD

**Issue by markrietveld** [bluelinelabs/LoganSquare#19]
It's standard behavior for other json parsers to skip fields with `static` and `transient` 
See a small model class below, with its generated code.

```java
import com.bluelinelabs.logansquare.annotation.JsonIgnore;
import com.bluelinelabs.logansquare.annotation.JsonObject;

@JsonObject(fieldDetectionPolicy = JsonObject.FieldDetectionPolicy.NONPRIVATE_FIELDS)
public class Test {
    public transient String transientString;
    public static final String staticString;
    public static final Parcelable.Creator<Test> CREATOR = new Parcelable.Creator<Test>() {
        @Override
        public Test createFromParcel(Parcel source) {
            return new Test();
        }

        @Override
        public Test[] newArray(int size) {
            return new Test[size];
        }
    };
}
```

generated code:
```java
public static void parseField(Test instance, String fieldName, JsonParser jsonParser) throws IOException {
    if ("transientString".equals(fieldName)) {
      instance.transientString = jsonParser.getValueAsString(null);
    } else if ("staticString".equals(fieldName)){
      instance.staticString = jsonParser.getValueAsString(null);
    } else if ("CREATOR".equals(fieldName)){
      instance.CREATOR = LoganSquare.typeConverterFor(android.os.Parcelable.Creator<Test>.class).parse(jsonParser);
    }
  }

public static void _serialize(Test object, JsonGenerator jsonGenerator, boolean writeStartAndEnd) throws IOException {
    if (writeStartAndEnd) {
      jsonGenerator.writeStartObject();
    }
    jsonGenerator.writeStringField("transientString", object.transientString);
    jsonGenerator.writeStringField("staticString", object.staticString);
    if (object.CREATOR != null) {
      LoganSquare.typeConverterFor(Parcelable.Creator<Test>.class).serialize(object.CREATOR, "CREATOR", true, jsonGenerator);
    }
    if (writeStartAndEnd) {
      jsonGenerator.writeEndObject();
    }
  }
```

**Comment by EricKuck** [comment]
Good catch. Unless something's specifically marked as a @JsonField, it doesn't make sense to parse static or transient variables. Will be fixed soon.

**Comment by EricKuck** [comment]
This is fixed in commit 853247c66e0f83069c3339eb93f09ffa77dfac23. I'll upload it to JCenter and Maven Central as version 1.0.4 before the end of the day.

**Comment by EricKuck** [comment]
1.0.4 has been pushed to JCenter and Maven Central, so you should be good to go now. Sorry about the issue!
